Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselItemAutomationPeer.cs

```diff
@@ -0,0 +1,217 @@
+        {
+            return AutomationControlType.ListItem;
+        }
+
+        /// <summary>
+        /// Called by GetClassName that gets a human readable name that, in addition to AutomationControlType,
+        /// differentiates the control represented by this AutomationPeer.
+        /// </summary>
+        /// <returns>The string that contains the name.</returns>
+        protected override string GetClassNameCore()
+        {
+            return Owner.GetType().Name;
+        }
+
+        /// <summary>
+        /// Called by GetName.
+        /// </summary>
+        /// <returns>
+        /// Returns the first of these that is not null or empty:
+        /// - Value returned by the base implementation
+        /// - Name of the owning CarouselItem
+        /// - Carousel class name
+        /// </returns>
+        protected override string GetNameCore()
+        {
+            string name = AutomationProperties.GetName(this.OwnerCarouselItem);
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
+
+            name = this.OwnerCarouselItem.Name;
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
+
+            var textBlock = this.OwnerCarouselItem.FindDescendant<TextBlock>();
+            if (textBlock != null)
+            {
+                return textBlock.Text;
+            }
+
+            name = base.GetNameCore();
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Called by GetAutomationId that gets the **AutomationId** of the element that is associated with the automation peer.
+        /// </summary>
+        /// <returns>
+        /// The string that contains the automation ID.
+        /// </returns>
+        protected override string GetAutomationIdCore()
+        {
+            var automationId = base.GetAutomationIdCore();
+            if (!string.IsNullOrEmpty(automationId))
+            {
+                return automationId;
+            }
+
+            if (this.OwnerCarouselItem != null)
+            {
+                return this.GetNameCore();
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Gets the control pattern that is associated with the specified Windows.UI.Xaml.Automation.Peers.PatternInterface.
+        /// </summary>
+        /// <param name="patternInterface">A value from the Windows.UI.Xaml.Automation.Peers.PatternInterface enumeration.</param>
+        /// <returns>The object that supports the specified pattern, or null if unsupported.</returns>
+        protected override object GetPatternCore(PatternInterface patternInterface)
+        {
+            switch (patternInterface)
+            {
+                case PatternInterface.SelectionItem:
+                    return this;
+            }
```

Comment:
Why not an if case?

Reply:
Could be changed. Wanted to keep this in line with the other automation peers which were created for the DataGrid in the toolkit so as not to cause any confusion in code styling.

Reply:
I would argue that an if case would be better in either case since a few interfaces all result in the same behavior. Having a switch case with 4 cases that all do the same (return `this`) looks a bit off in my opinion, but that's just my opinion.